zio-http/src/main/scala/zhttp/http/middleware/package.scala

```diff
@@ -0,0 +1,6 @@
+package zhttp.http
+
+package object middleware {
+  type HttpMiddleware[-R, +E] = Middleware[R, E, Request, Response, Request, Response]
+  type RequestP[+A]           = (Method, URL, Headers) => A
```

Comment:
We can create a new type like below instead of `RequestP`

```scala
case class MiddlewareRequest(
  private val request: Request
) extends HeaderExtensions[MiddlewareRequest] {
 // ...

  def method: Method = ???
  def url: URL = ???
  def headers: Headers = ???
}
```

Using `MiddlewareRequest` instead of passing a bunch of params everywhere.

Reply:
https://github.com/dream11/zio-http/pull/798